### ISSUE TITLE
[Regression BUG] Class scoped function showing SyntaxError:  'this' is not allowed before super()

### ISSUE THREAD

**Issue by ldong** [babel/babel#2971]
Code Example

```javascript
class Parent {
    constructor() {
  }
};

class Child extends Parent {
    constructor() {
        super();
    }
    
    scopedFunctionWithThis = () => {
        this.name = {};
    }
}
```
1. Save above snippet as `a.js`
2. Run `babel --presets stage-0  a.js`, 
3. Error Message: `SyntaxError: a.js: 'this' is not allowed before super() (This is an error on an internal node. Probably an internal error)` 

This is fine on [babel repl](https://babeljs.io/repl/#?experimental=true&evaluate=true&loose=false&spec=false&code=class%20Parent%20%7B%0A%20%20%20%20constructor()%20%7B%0A%20%20%7D%0A%7D%3B%0A%0Aclass%20Child%20extends%20Parent%20%7B%0A%20%20%20%20constructor()%20%7B%0A%20%20%20%20%20%20%20%20super()%3B%0A%20%20%20%20%7D%0A%20%20%20%20%0A%20%20%20%20scopedFunctionWithThis%20%3D%20()%20%3D%3E%20%7B%0A%20%20%20%20%20%20%20%20this.name%20%3D%20%7B%7D%3B%0A%20%20%20%20%7D%0A%7D%0A) however.

Environment:
* node: v4.2.1
* npm: 2.14.7
* Packages: babel-core, es2015, stage0 are installed

**Comment by julienvincent** [comment]
Was about to open an issue for this as well. I'm getting this too.

**Comment by UchihaVeha** [comment]
+1 i'm getting this too

**Comment by williamboman** [comment]
+1 https://github.com/williamboman/babel-errors/tree/this-before-super

**Comment by thejameskyle** [comment]
Duplicate of #2942

**Comment by aaronbeall** [comment]
@thejameskyle What is required for the fix? I'm using babel-core 6.1.19, babel-preset-es2015 6.1.18, babel-preset-react 6.1.18, babel-plugin-transform-class-properties 6.1.18 and still seeing this problem *only with arrow function properties* (not with other class properties).
